util/middleware.js

```diff
@@ -9,38 +9,59 @@ export function getUserUrl (type) {
 }
 
 /**
- * This convenience method wraps a fetch call to the specified URL
- * with the token and api key added (if provided) to the HTTP request header.
+ * This method wraps a fetch call to the specified URL
+ * with the auth0 object and api key added (if provided) to the HTTP request header.
+ * In case of an error obtaining a token, this method returns a response that
+ * mimics the fetch response for an unauthorized request (HTTP 401).
  * @param {string} url The URL to call.
- * @param {string} accessToken If non-null, the Authorization token to add to request header.
+ * @param {string} auth0 the auth0 object used to obtain an accessToken for secure APIs.
  * @param {string} method The HTTP method to execute.
  * @param {*} options Extra fetch options to pass to fetch.
  */
-export async function secureFetch (url, accessToken, method = 'get', options = {}) {
-  const res = await fetch(url, {
+async function secureFetchInternal (url, auth0, method = 'get', options = {}) {
+  const { getAccessTokenSilently } = auth0
+  let accessToken
+  try {
+    accessToken = await getAccessTokenSilently()
```

Comment:
Can you add the note here about successive calls to Auth0 not being made?

Reply:
Yes. Updated in https://github.com/ibi-group/otp-admin-ui/pull/33/commits/bf8b05f859ef90a687cb2f6ca2bc8bf2d3ec4938.